src/main/java/ro/utcn/sd/alexh/assignment1/persistence/jdbc/JdbcAnswerRepository.java

```diff
@@ -0,0 +1,79 @@
+package ro.utcn.sd.alexh.assignment1.persistence.jdbc;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import ro.utcn.sd.alexh.assignment1.entity.Answer;
+import ro.utcn.sd.alexh.assignment1.persistence.api.AnswerRepository;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+public class JdbcAnswerRepository implements AnswerRepository {
+
+    private final JdbcTemplate template;
+
+    @Override
+    public Answer save(Answer answer) {
+        if (answer.getAnswerId() != null) {
+            update(answer);
+        } else {
+            Integer id = insert(answer);
+            answer.setAnswerId(id);
+        }
+        return answer;
+    }
+
+    @Override
+    public Optional<Answer> findById(int id) {
+        List<Answer> answers = template.query("SELECT * FROM answer WHERE answer_id = ?",
+                (resultSet, i) -> new Answer(
```

Comment:
[Optional] To reduce the duplicate coding, you could define this RowMapper as a separate class. It could even be a nested class (defined inside this file).